kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/KafkaPublisher.java

```diff
@@ -123,37 +124,42 @@ protected KafkaPublisher(Builder<K, V> builder) {
      * @param event the events to publish on the Kafka broker.
      * @param <T>   the implementation of {@link EventMessage} send through this method
      */
+    @SuppressWarnings("squid:S2095") //producer needs to be closed async, not within this method
     public <T extends EventMessage<?>> void send(T event) {
         logger.debug("Starting event producing process for [{}].", event.getPayloadType());
+        Optional<String> topic = topicResolver.apply(event);
```

Comment:
We could provide a method on the `TopicResolver` called `resolve`. 
That makes it clearer from this piece of code.

The implementation. although boring, could look a little bit like this:
```java
default Optional<String> resolve(EventMessage<?> event) {
    return this.apply(event);
}
```